tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs

```diff
@@ -0,0 +1,166 @@
+    {
+        // Arrange
+        var command = new ApproveBorrowRequest.Command()
+        {
+            BorrowId = Guid.NewGuid(),
+        };
+        
+        // Act
+        var result = async () => await SendAsync(command);
+
+        // Assert
+        await result.Should().ThrowAsync<KeyNotFoundException>()
+            .WithMessage("Borrow request does not exist.");
+    }
+
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenDocumentIsLost()
+    {
+        // Arrange
+        var document = CreateNDocuments(1).First();
+
+        document.Status = DocumentStatus.Lost;
+
+        var user = CreateUser(IdentityData.Roles.Employee, "abcdef");
+
+        var request = CreateBorrowRequest(user, document, BorrowRequestStatus.Pending);
+
+        await AddAsync(request);
+        
+        var command = new ApproveBorrowRequest.Command()
+        {
+            BorrowId = request.Id,
+        };
+        
+        // Act
+        var result = async () => await SendAsync(command);
+        
+        // Assert
+        await result.Should().ThrowAsync<ConflictException>()
+            .WithMessage("Document is lost. Request is unprocessable.");
+        
+        // Cleanup
+        Remove(request);
+        Remove(user);
+        Remove(document);
+    }
+
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenRequestStatusIsNotPendingAndRejected()
+    {
+        var document = CreateNDocuments(1).First();
+
+        var user = CreateUser(IdentityData.Roles.Employee, "abcdef");
+
+        var request = CreateBorrowRequest(user, document, BorrowRequestStatus.CheckedOut);
+
+        await AddAsync(request);
+        
+        var command = new ApproveBorrowRequest.Command()
+        {
+            BorrowId = request.Id,
+        };
+        
+        // Act
+        var result = async () => await SendAsync(command);
+        
+        // Assert
+        await result.Should().ThrowAsync<ConflictException>()
+            .WithMessage("Request cannot be approved.");
+        
+        // Cleanup
+        Remove(request);
+        Remove(user);
+        Remove(document);
+    }
+
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenRequestTimespanOverlapAnApprovedOrCheckedOutRequestTimespan()
+    {
+        using var scope = ScopeFactory.CreateScope();
+        var context = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+        var document = CreateNDocuments(1).First();
+
+        var user1 = CreateUser(IdentityData.Roles.Employee, "abcdef");
+        var user2 = CreateUser(IdentityData.Roles.Employee, "aaaaaa");
+
+        var request1 = CreateBorrowRequest(user1, document, BorrowRequestStatus.Approved);
+        
+        var request2 = CreateBorrowRequest(user2, document, BorrowRequestStatus.Pending);
+        request2.BorrowTime = request2.BorrowTime.Plus(Period.FromHours(1));
+        request2.BorrowTime = request2.BorrowTime.Plus(Period.FromHours(1));
```

Comment:
you forgot to change to request1

Reply:
clarification, the second line should have changed the due time of the request, but i forgot

Reply:
reasonable